react-hash-routing/client/app.jsx

```diff
@@ -0,0 +1,42 @@
+import React, { useEffect, useState } from 'react';
+import Header from './components/header';
+import Catalog from './pages/catalog';
+import NotFound from './pages/not-found';
+import ProductDetails from './pages/product-details';
+import { parseRoute } from './lib';
+
+export default function App() {
+  const [route, setRoute] = useState(parseRoute(window.location.hash));
+
+  useEffect(() => {
+    /**
+     * Listen for hash change events on the window object
+     * Each time the window.location.hash changes, parse
+     * it with the parseRoute() function and update state
+     * Clean up when the component unmounts.
+     */
+    window.addEventListener('hashchange', () => {
+      const route = parseRoute(window.location.hash);
+      setRoute(route);
+    });
+    return () => window.removeEventListener('hashchange');
```

Comment:
This doesn't quite work. To remove an event listener you have to pass in the same function you passed in when you added the event listener.